src/hooks/useAdvanceStep.js

```diff
@@ -0,0 +1,30 @@
+import { useState } from 'react'
+
+function useAdvanceStep() {
+  const [currentStep, setCurrentStep] = useState(0)
+
+  const scrollTop = () => window.scrollTo({ x: 0 })
+  const changeTitle = title => (document.title = title)
+  const setFocus = () => {
+    setTimeout(() => {
+      const focusTarget = document.getElementById('focus-target')
```

Comment:
Did you try passing a ref to the hook for this component instead of querying from the document? I always thought this is a anti pattern in react.

Maybre try to pass the form as ref here :)

Reply:
I'll look into refs. Haven't really worked with them before and also felt a bit weird to do it like this as I don't like to call the `document` directly in React.